tests/kafkatest/tests/streams/streams_upgrade_test.py

```diff
@@ -454,6 +470,11 @@ def do_stop_start_bounce(self, processor, upgrade_from, new_version, counter):
                                            timeout_sec=60,
                                            err_msg="Never saw output '%s' on " % self.processed_data_msg + str(node.account))
 
+                        if "test.run_fk_join" in extra_properties:
```

Comment:
nit: let's also check that the corresponding value is `True`

```suggestion
                        if extra_properties.get('test.run_fk_join', False):
```